### PR TITLE
Add "document-before-save-as" signal

### PR DESCRIPTION
**Pull request by techee**
Right now only the "document-save" is fired when performing "save as" which lacks the information about the original file name (which, for instance, is necessary for the LSP client where it needs to notify the server that the "old" document was closed and the document under the new name opened).

Right now the plugin at https://github.com/techee/geany-lsp has the code using this signal commented-out - I'll uncomment it once it's added to Geany (so renaming a file might cause problems synchronizing the document with the server with the current implementation).

### CODE REVIEW

File: doc/pluginsignals.c [hunk 1]
```diff
@@ -92,6 +92,17 @@ signal void (*document_reload)(GObject *obj, GeanyDocument *doc, gpointer user_d
...
+/** Sent before save as is performed with the original document.
+ *
+ * @param obj a GeanyObject instance, should be ignored.
+ * @param doc the original document. The document with the new file name is still
+ *        reported by the "document-save" signal sent afterwards.
+ * @param user_data user data.
+ *
+ * @since TODO
+ */
+signal void (*document_before_save_as)(GObject *obj, GeanyDocument *doc, gpointer user_data);
```
Comment (kugel-) [inline doc/pluginsignals.c:104]: Perhaps add a `boolean` return value to potentially allow the plugin to block saving the file?

Comment (techee) [reply doc/pluginsignals.c:104]: Hmm, I'm not sure if plugins should be allowed to do that and what exactly would this be good for. Do you have some example?

Comment (elextr) [reply doc/pluginsignals.c:104]: No, the plugin should not block a user saving a file, under any circumstances.

**Comment by elextr** [comment]
Looks ok to me but I know nothing about GTK objects, but it looks the same as document_before_save :-)

**Review by b4n — APPROVED** [review]
LGTM

**Pull request merged**